src/store/edit_post_form.js

```diff
@@ -29,7 +29,7 @@ const initialState = i.fromJS({
 export default function reducer(state = initialState, action) {
   switch (action.type) {
     case posts.RESET_EDIT_POST_FORM: {
-      state.set(action.post_id, i.fromJS({
+      state.set(action.payload.post_id, i.fromJS({
```

Comment:
There's an issue on save while editing posts:
```
Unhandled rejection TypeError: Cannot read property 'post_id' of undefined
    at reducer (http://localhost:8000/assets/app.js:88675:33)
    at http://localhost:8000/assets/app.js:213687:22
    at http://localhost:8000/assets/app.js:4491:46
    at http://localhost:8000/assets/app.js:2852:16
    at ValueNode.iterate (http://localhost:8000/assets/app.js:3219:12)
    at BitmapIndexedNode.iterate.HashArrayMapNode.iterate (http://localhost:8000/assets/app.js:3212:24)
    at HashArrayMapNode.BitmapIndexedNode.iterate.HashArrayMapNode.iterate (http://localhost:8000/assets/app.js:3212:24)
    at Map.__iterate (http://localhost:8000/assets/app.js:2850:32)
    at KeyedIterable.mappedSequence.__iterateUncached (http://localhost:8000/assets/app.js:4490:23)
    at seqIterate (http://localhost:8000/assets/app.js:2079:16)
    at KeyedIterable.Seq.__iterate (http://localhost:8000/assets/app.js:1749:14)
    at KeyedIterable.forEach (http://localhost:8000/assets/app.js:5856:19)
    at http://localhost:8000/assets/app.js:2697:16
    at Map.withMutations (http://localhost:8000/assets/app.js:2828:7)
    at new Map (http://localhost:8000/assets/app.js:2694:20)
    at reify (http://localhost:8000/assets/app.js:5045:37)
From previous event:
    at http://localhost:8000/assets/app.js:66048:99
    at http://localhost:8000/assets/app.js:66195:21
    at Object.ReactErrorUtils.invokeGuardedCallback (http://localhost:8000/assets/app.js:44866:16)
    at executeDispatch (http://localhost:8000/assets/app.js:44375:21)
    at Object.executeDispatchesInOrder (http://localhost:8000/assets/app.js:44398:5)
    at executeDispatchesAndRelease (http://localhost:8000/assets/app.js:34891:22)
    at executeDispatchesAndReleaseTopLevel (http://localhost:8000/assets/app.js:34902:10)
    at Array.forEach (native)
    at forEachAccumulated (http://localhost:8000/assets/app.js:58454:9)
    at Object.processEventQueue (http://localhost:8000/assets/app.js:35105:7)
    at runEventQueueInBatch (http://localhost:8000/assets/app.js:125810:18)
    at Object.handleTopLevel [as _handleTopLevel] (http://localhost:8000/assets/app.js:125821:5)
    at handleTopLevelImpl (http://localhost:8000/assets/app.js:125908:24)
    at ReactDefaultBatchingStrategyTransaction.perform (http://localhost:8000/assets/app.js:38464:20)
    at Object.batchedUpdates (http://localhost:8000/assets/app.js:125651:26)
    at Object.batchedUpdates (http://localhost:8000/assets/app.js:27688:27)
    at dispatchEvent (http://localhost:8000/assets/app.js:125983:20)printWarning @ bluebird.js:1542formatAndLogError @ bluebird.js:1258fireRejectionEvent @ bluebird.js:1283Promise._notifyUnhandledRejection @ bluebird.js:729(anonymous function) @ bluebird.js:154
```

If you go this way, you should put a `payload` with `post_id` in `resetEditPostForm()`: [`posts.js:106`](https://github.com/Lokiedu/libertysoil-site/blob/feature/fsa/src/actions/posts.js#L106).

But instead why don't just throw away `state.set(action.payload.post_id, i.fromJS({ ... ` and just enter `state = initialState` if this action should really *reset* the form?

___

I catch another issue with the version of line: it doesn't update the state because it's immutable.